packages/malloy-db-trino/src/trino_connection.ts

```diff
@@ -276,42 +225,43 @@ export abstract class TrinoPrestoConnection
     return data as unknown[];
   }
 
-  convertRow(structDef: StructDef, _row: unknown) {
+  convertRow(structDef: StructDef, rawRow: unknown) {
     const retRow = {};
-    const row = this.unpackArray(_row);
+    const row = this.unpackArray(rawRow);
     for (let i = 0; i < structDef.fields.length; i++) {
       const field = structDef.fields[i];
 
-      if (field.type === 'struct') {
-        const struct = field as StructDef;
-        if (struct.structSource.type === 'inline') {
-          retRow[field.name] = this.convertRow(struct, row[i]);
-        } else {
-          retRow[field.name] = this.convertNest(struct, row[i]);
-        }
+      if (field.type === 'record') {
+        retRow[field.name] = this.convertRow(field, row[i]);
+      } else if (isRepeatedRecord(field)) {
+        // mtoy todo if you don't have tests which end up testing this, test this
+        throw new Error(
+          "Trino dialect can't convert a repeated record in a result"
+        );
+        // retRow[field.name] = this.convertNest(field, row[i]);
+      } else if (field.type === 'array') {
+        // mtoy todo fail a test then fix this too
+        throw new Error(
+          "Trino dialect can't convert an array in the result set"
+        );
+        // retRow[field.name] = this.convertNest(field, row[i]);
       } else {
-        retRow[field.name] = row[i] === undefined ? null : row[i];
+        retRow[field.name] = row[i] ?? null;
       }
     }
     //console.log(retRow);
     return retRow;
   }
 
-  convertNest(structDef: StructDef, _data: unknown) {
-    const data = this.unpackArray(_data);
-    const ret: unknown[] = [];
-    //console.log(
-    //   `${JSON.stringify(structDef, null, 2)} ${JSON.stringify(data, null, 2)} `
-    // );
-    if (structDef.structSource.type === 'inline') {
-      return this.convertRow(structDef, data);
-    }
-    const rows = (data === null || data === undefined ? [] : data) as unknown[];
-    for (const row of rows) {
-      ret.push(this.convertRow(structDef, row));
-    }
-    return ret;
-  }
+  // convertNest(structDef: StructDef, _data: unknown) {
+  //   const data = this.unpackArray(_data);
+  //   const ret: unknown[] = [];
+  //   const rows = (data === null || data === undefined ? [] : data) as unknown[];
+  //   for (const row of rows) {
+  //     ret.push(this.convertRow(structDef, row));
+  //   }
+  //   return ret;
+  // }
```

Comment:
Delete?

Reply:
Actually awaiting the other TODO because I decide to re-write or delete

Reply:
fixed